### PR TITLE
DDST-382: Subscriber to handle deletion of related Embargoes when a n…

### PR DESCRIPTION
**Pull request by Prashant-bd**
…ode is deleted

### CODE REVIEW

File: src/IslandoraNodeDeleteSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  /**
+   * Deletes the associated embargos.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The Islandora object node being deleted.
+   */
+  public function deleteAssociatedCustomEntity(NodeInterface $node): void {
+    $custom_entity_storage = $this->entityTypeManager->getStorage('embargo');
```
Comment (adam-vessey) [inline src/IslandoraNodeDeleteSubscriber.php:37]: This would effectively be creating a dependency on the `embargo` module, yeah? Probably not desired.

**Review by adam-vessey — COMMENTED** [review]
Not entirely sure this is desirable, and definitely not in some other module? Like, if this behaviour is expected, it should probably be implemented in embargo proper.

File: src/IslandoraNodeDeleteSubscriber.php (hunk 1 above)
Comment (Prashant-bd) [reply src/IslandoraNodeDeleteSubscriber.php:37]: @adam-vessey please suggest in which module I can implement this. https://github.com/discoverygarden/embargo module?

Comment (adam-vessey) [reply src/IslandoraNodeDeleteSubscriber.php:37]: Would be more of the target, yes. Still not convinced that such functionality is desired.

Like, something like a scan to find occurrences of `::getEmbargoedNode()` that don't account for it being able to be `NULL` might be more in order?

The ticket is more about error suppression than data maintenance that this PR is getting into, or so I read it. As in, much like https://github.com/discoverygarden/embargo/pull/41 , might also need to hit: https://github.com/discoverygarden/embargo/blob/a075aa4a994472666adf9cdd1af48891fb675323/src/Plugin/search_api/processor/EmbargoJoinProcessor.php#L230

**Pull request closed without merging**